@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -34,23 +48,92 @@ export interface TileGeometryLoader {
      */
     allGeometryLoaded: boolean;
 
+    /**
+     * The kinds of geometry stored in this [[Tile]].
+
+     */
+    availableGeometryKinds: GeometryKindSet | undefined;
+
     /**
      * Start with or continue with loading geometry. Called repeatedly until `isFinished` is `true`.
      */
-    update(): void;
+    update(
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined
+    ): void;
 
     /**
      * Dispose of any resources.
      */
     dispose(): void;
 }
 
+export namespace TileGeometryLoader {
+    /**
+     * Make sure that all technique have their geometryKind set, either from the theme or their
+     * default value.
+     *
+     * Also gather set of the [[GeometryKind]]s found in the techniques and return it.
+     *
+     * @param {DecodedTile} decodedTile
+     * @returns {GeometryKindSet} The set of kinds used in the decodeTile.
+     */
+    export function prepareDecodedTile(decodedTile: DecodedTile): GeometryKindSet {
+        const foundSet: GeometryKindSet = new GeometryKindSet();
+
+        for (const technique of decodedTile.techniques) {
```

Comment:
Same block of code is in PhasedTileGeometryLoader. Can we remove duplicate code somehow?

Reply:
"Yes 'we' can"!